pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -86,11 +86,13 @@ func SetDefaults_SeedNetworks(obj *SeedNetworks) {
 
 // SetDefaults_SeedSettingDependencyWatchdog sets defaults for SeedSettingDependencyWatchdog objects.
 func SetDefaults_SeedSettingDependencyWatchdog(obj *SeedSettingDependencyWatchdog) {
-	if obj.Endpoint == nil {
-		obj.Endpoint = &SeedSettingDependencyWatchdogEndpoint{Enabled: true}
+	// TODO(himanshu-kun): update once deprecated fields are removed
+	// If any of the new or deprecated fields is set, then syncing logic should take care, as this runs before that.
+	if obj.Weeder == nil && obj.Endpoint == nil {
+		obj.Weeder = &SeedSettingDependencyWatchdogWeeder{Enabled: true}
 	}
-	if obj.Probe == nil {
-		obj.Probe = &SeedSettingDependencyWatchdogProbe{Enabled: true}
+	if obj.Prober == nil && obj.Probe == nil {
+		obj.Prober = &SeedSettingDependencyWatchdogProber{Enabled: true}
 	}
 }
```

Comment:
It seems this code can never be executed since we will never hit a case where both `Weeder=Endpoint=nil` after https://github.com/gardener/gardener/pull/6693/files#diff-5f77dc0104e4ba123994ddc0cf72e5c0e3e166c70d11026084c4791c041ff953R73. Let's remove all defaulting code for the `SeedSettingDependencyWatchdog` field here and only have the code in the registry for the time being (until the deprecated `Endpoint`/`Probe` fields get removed)